test/functional/kernel/hyperplane/tests/test-kernel-hyperplane-2D.hpp

```diff
@@ -11,39 +11,72 @@
 #include <numeric>
 #include <type_traits>
 
-template <typename INDEX_TYPE, typename DATA_TYPE, typename WORKING_RES, typename EXEC_POLICY, typename REDUCE_POLICY>
-void KernelHyperplane2DTestImpl(const int groups, const int idim, const int jdim)
-{
-  // This test traverses "groups" 2D arrays, and modifies values in a 1D hyperplane manner.
+template<typename DATA_TYPE, typename REDUCE_POLICY, bool TypeTrait>
+struct ReducerHelper {};
 
-  camp::resources::Resource work_res{WORKING_RES::get_default()};
+template<typename DATA_TYPE, typename REDUCE_POLICY>
+struct ReducerHelper<DATA_TYPE, REDUCE_POLICY, false> {
+  using type = RAJA::ReduceSum<REDUCE_POLICY, DATA_TYPE>;
+};
 
-  DATA_TYPE * work_array;
-  DATA_TYPE * check_array;
-  DATA_TYPE * test_array;
+template<typename DATA_TYPE, typename REDUCE_POLICY>
+struct ReducerHelper<DATA_TYPE, REDUCE_POLICY, true> {
+  using type = DATA_TYPE;
+};
 
-  INDEX_TYPE array_length = groups * idim * jdim;
+template <typename INDEX_TYPE, typename DATA_TYPE, typename EXEC_POLICY, typename REDUCE_POLICY, typename USE_PARAM_REDUCER>
+typename std::enable_if<USE_PARAM_REDUCER::value>::type
```

Comment:
use cpp14 std::..._t<...> aliases for brevity.
```suggestion
std::enable_if_t<USE_PARAM_REDUCER::value>
```